nemo/collections/asr/parts/feature_loader.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pickle
+
+import numpy as np
+import torch
+
+
+class ExternalFeatureLoader(object):
+    def __init__(self, file_path, sample_rate=16000, int_values=False, augmentor=None):
+        self._file_path = file_path
+        self.augmentor = augmentor
+
+    def load_feature_from_file(self, file_path):
+        # load pickle file
+        if file_path.endswith(".p") or file_path.endswith(".pickle"):
```

Comment:
Why .p ? The standard in python is .pt for pytorch, .npy for numpy and .pkl for pickle objects. Also don't make it so brittle as to check filepath extension, there's many cases where pickle files are written with different extension.

Reply:
Good catch. It should be .pickle or .pkl..... .p is too confusing. 
Sure will have a smarter way to check and open pickle file.

Reply:
ok `np.load(..., allow_pickle=True)` is a good way to take care of `pkl, npz, and npy`  at the same time!